python/wpt/grouping_formatter.py

```diff
@@ -112,14 +112,22 @@ class ServoHandler(mozlog.reader.LogHandler):
     """LogHandler designed to collect unexpected results for use by
     script or by the ServoFormatter output formatter."""
 
-    def __init__(self):
+    def __init__(self, detect_flakes=False):
+        """
+        Flake detection assumes first suite is actual run
+        and rest of the suites are retry-unexpected for flakes detection.
+        """
+        self.detect_flakes = detect_flakes
         self.reset_state()
 
     def reset_state(self):
         self.number_of_tests = 0
         self.completed_tests = 0
         self.need_to_erase_last_line = False
         self.running_tests: Dict[str, str] = {}
+        if self.currently_detecting_flakes:
+            return
+        self.currently_detecting_flakes = False
```

Comment:
Oh, this may need to be set in `__init__` instead of here so that it is initialized before the if-statement above.